### PR TITLE
apiv2(ticdc): implement the list processors

### PR DESCRIPTION
**Pull request by charleszheng44**
### What problem does this PR solve?

Issue Number: ref #8019 

### What is changed and how it works?

Implement the list processors for API v2

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No

##### Do you need to update user documentation, design documentation or monitoring documentation?
No

### Release note 
```release-note
None
```

### CODE REVIEW

**Review by sdojjy — APPROVED** [review]

File: cdc/api/v2/processor.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+// listProcessors lists all processors in the TiCDC cluster
+// @Summary List processors
+// @Description list all processors in the TiCDC cluster
+// @Tags processor
+// @Accept json
+// @Produce json
+// @Success 200 {array} model.ProcessorCommonInfo
+// @Failure 500,400 {object} model.HTTPError
+// @Router	/api/v1/processors [get]
```
Comment (sdojjy) [inline cdc/api/v2/processor.go:31]:
```suggestion
// @Router	/api/v2/processors [get]
```

File: cdc/api/v2/processor.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/tiflow/cdc/model"
+)
+
+// listProcessors lists all processors in the TiCDC cluster
+// @Summary List processors
+// @Description list all processors in the TiCDC cluster
+// @Tags processor
+// @Accept json
```
Comment (sdojjy) [inline cdc/api/v2/processor.go:27]: remove this line

File: cdc/api/v2/processor.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+// @Failure 500,400 {object} model.HTTPError
+// @Router	/api/v1/processors [get]
+func (h *OpenAPIV2) listProcessors(c *gin.Context) {
+	ctx := c.Request.Context()
+	infos, err := h.capture.StatusProvider().GetProcessors(ctx)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	resps := make([]*model.ProcessorCommonInfo, len(infos))
```
Comment (sdojjy) [inline cdc/api/v2/processor.go:39]: follow the API style, use `ListResponse`

File: cdc/api/v2/processor.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	for i, info := range infos {
+		resp := &model.ProcessorCommonInfo{
+			Namespace: info.CfID.Namespace,
+			CfID:      info.CfID.ID, CaptureID: info.CaptureID,
```
Comment (sdojjy) [inline cdc/api/v2/processor.go:43]:
```suggestion
			CfID:           info.CfID.ID, 
			CaptureID: info.CaptureID,
```

File: cdc/api/v2/api.go [hunk 5]
```diff
@@ -63,6 +63,11 @@ func RegisterOpenAPIV2Routes(router *gin.Engine, api OpenAPIV2) {
 	captureGroup.POST("/:capture_id/drain", api.drainCapture)
 	captureGroup.GET("", api.listCaptures)
 
+	// processor apis
+	processorGroup := v2.Group("/processors")
+	processorGroup.Use(middleware.ForwardToOwnerMiddleware(api.capture))
+	captureGroup.GET("", api.listProcessors)
```
Comment (sdojjy) [inline cdc/api/v2/api.go:69]:
```suggestion
	processorGroup.GET("", api.listProcessors)
```

File: cdc/api/v2/processor.go (hunk 2 above)
Comment (charleszheng44) [reply cdc/api/v2/processor.go:27]: removed

File: cdc/api/v2/processor.go (hunk 1 above)
Comment (charleszheng44) [reply cdc/api/v2/processor.go:31]: fixed

File: cdc/api/v2/processor.go (hunk 4 above)
Comment (charleszheng44) [reply cdc/api/v2/processor.go]: fixed

File: cdc/api/v2/api.go (hunk 5 above)
Comment (charleszheng44) [reply cdc/api/v2/api.go]: fixed

File: cdc/api/v2/processor.go (hunk 3 above)
Comment (charleszheng44) [reply cdc/api/v2/processor.go]: changed to `ListResponse`

**Review by sdojjy — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**